src/browser_launchers/browser_launcher.js

```diff
@@ -0,0 +1,118 @@
+import ow from 'ow';
+import { ENV_VARS } from 'apify-shared/consts';
+import { getTypicalChromeExecutablePath } from '../utils';
+
+export default class BrowserLauncher {
+    static optionsShape = {
+        launcher: ow.optional.object, // @TODO: I have dropped support for string, since it is not supported by the playwright.
+        proxyUrl: ow.optional.string.url,
+        useChrome: ow.optional.boolean,
+        launchOptions: ow.optional.object,
+    }
+
+    constructor(launchContext) {
+        ow(launchContext, 'BrowserLauncherOptions', ow.object.exactShape(BrowserLauncher.optionsShape));
+
+        const {
+            launcher,
+            proxyUrl,
+            useChrome,
+            launchOptions = {},
+        } = launchContext;
+
+        this._validateProxyUrl(proxyUrl);
+
+        this.launcher = launcher;
+        this.proxyUrl = proxyUrl;
+        this.useChrome = useChrome;
+        this.launchOptions = launchOptions;
+
+        this.Plugin = null; // tight to specific library implementation;
```

Comment:
I'd say "to be provided by child classes" or something similar. The comment was not very clear to me at first.